web3/middleware/signing.py

```diff
@@ -0,0 +1,92 @@
+from eth_account.local import (
+    LocalAccount,
+)
+from eth_keys.datatypes import (
+    PrivateKey,
+)
+from eth_utils import (
+    is_same_address,
+)
+from toolz import (
+    compose,
+)
+
+from web3.utils.formatters import (
+    apply_formatter_if,
+)
+from web3.utils.transactions import (
+    fill_nonce,
+    fill_transaction_defaults,
+)
+
+to_hexstr_from_eth_key = lambda x: x.to_hex()  # noqa: E731
+
+
+def is_eth_key(value):
+    if isinstance(value, PrivateKey):
+        return True
+    return False
+
+
+def raw_key_formatter(private_key):
+    formatter = compose(
+        apply_formatter_if(is_eth_key, to_hexstr_from_eth_key),
+    )
+    return formatter(private_key)
+
+
+def to_account(web3, private_key):
```

Comment:
We've been using the convention of `w3` as the instance name for web3 instances.